File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/visit/VisitHistoryCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.visit;

import java.time.LocalDate;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;
import com.depromeet.threedollar.domain.rds.user.domain.store.Store;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class VisitHistoryCreator {

    @Builder
    public static VisitHistory create(
        @NotNull Store store,
        @NotNull Long userId,
        @NotNull VisitType type,
        @NotNull LocalDate dateOfVisit
    ) {
        return VisitHistory.builder()
            .store(store)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/visit/VisitHistoryCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.visit

import java.time.LocalDate
import com.depromeet.threedollar.domain.rds.domain.userservice.store.Store
import com.depromeet.threedollar.domain.rds.domain.userservice.visit.VisitHistory
import com.depromeet.threedollar.domain.rds.domain.userservice.visit.VisitType
import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object VisitHistoryCreator {

    @JvmStatic
    fun create(
        store: Store,
        userId: Long,
        type: VisitType,
        dateOfVisit: LocalDate,
    ): VisitHistory {
        return VisitHistory.builder()
            .store(store)
            .userId(userId)
            .type(type)
            .dateOfVisit(dateOfVisit)
            .build()
    }

}
